v2/tools/generator/internal/codegen/pipeline/convert_allof_and_oneof_to_objects.go

```diff
@@ -255,6 +255,13 @@ func (s synthesizer) getOneOfName(t astmodel.Type, propIndex int) (propertyNames
 			json:       s.idFactory.CreateIdentifier(name, astmodel.NotExported),
 			isGoodName: false, // TODO: This name sucks but what alternative do we have?
 		}, nil
+	case *astmodel.MapType:
```

Comment:
this was a missing case, which was triggered by this change. I'm not _entirely_ sure why, but plugging this hole.